app/services/cancellation_fee_calculator.rb

```diff
@@ -12,10 +12,10 @@ def fee
 
   def calculated_fee
     unless defined?(@calculated_fee)
-      original_canceled_at = @reservation.canceled_at
-      @reservation.canceled_at = Time.current
+      original_canceled_at = @reservation.order_detail.canceled_at
```

Comment:
Hmm, this is a spot where now I question the value of removing the delegate. Might be worth discussing.